### PR TITLE
Feat/#48 data

### PR DESCRIPTION
**Pull request by ciaranschutte**
adds api call to create a new application

### CODE REVIEW

File: components/pages/Applications/Dashboard/Applications/Start.tsx [hunk 1]
```diff
@@ -4,11 +4,27 @@ import Icon from '@icgc-argo/uikit/Icon';
...
 import React from 'react';
 import DashboardCard from '../Card';
+import { useRouter } from 'next/router';
 
 const StartApplication = () => {
   const theme = useTheme();
+  const { fetchWithAuth } = useAuthContext();
+  const router = useRouter();
+
+  const createNewApplication = () => {
```
Comment (ciaranschutte) [inline components/pages/Applications/Dashboard/Applications/Start.tsx:17]: call to create application on backend and then navigates on success.

in my right now our Application page will do a call with just the `appId`
on load the application page will another request for the same data we just got back.
we can optimise for this later i think.
there are some page level data/routing changes that need to be adjusted and it falls into this camp

File: components/pages/Applications/Dashboard/Applications/index.tsx [hunk 2]
```diff
@@ -1,19 +1,13 @@
...
-  inProgressApplication?: any;
-};
-
-const application = { appId: '', state: '', submitterId: '', expiresAtUtc: '', updatedAtUtc: '' };
-
-const Applications = ({ inProgressApplication = null }: ApplicationsProps) => {
-  const theme: UikitTheme = useTheme();
+const Applications = () => {
+  const { error, isLoading, response } = useApplicationsAPI({});
+  const inProgressApplications: any = []; //response?.data;
```
Comment (ciaranschutte) [inline components/pages/Applications/Dashboard/Applications/index.tsx:10]: small fix to place where logic is. no other work on this component in this PR. outside ticket scope

File: components/pages/Applications/types.ts [hunk 3]
```diff
@@ -18,9 +18,9 @@ export type SortedChangeFunction = (
 ) => void;
 
 export type ApplicationsRequestData = {
-  page: number;
-  pageSize: number;
-  sort: string;
+  page?: number;
```
Comment (ciaranschutte) [inline components/pages/Applications/types.ts:21]: make params optional as we added default params

File: package.json [hunk 4]
```diff
@@ -7,7 +7,8 @@
     "build": "next build",
     "start": "next start",
     "storybook": "start-storybook -p 6006",
-    "build-storybook": "build-storybook"
+    "build-storybook": "build-storybook",
+    "mock-server": "npx json-server --watch mock/db.json --port 3001"
```
Comment (ciaranschutte) [inline package.json:11]: add a mock server setup as I've added it a few times without ever committing. can remove later once we're more stable.

File: components/pages/Applications/Dashboard/Applications/InProgress/index.tsx [hunk 5]
```diff
@@ -6,11 +6,12 @@ import ProgressBar from '../../../../../ApplicationProgressBar';
 import { SIMPLE_DATE_FORMAT, TIME_AND_DATE_FORMAT } from './constants';
 import { getFormattedDate, getStatusText } from './helpers';
 import ButtonGroup from './ButtonGroup';
+import { ApplicationState } from 'components/ApplicationProgressBar/types';
 
 const initState = { appId: '', state: '', submitterId: '', expiresAtUtc: '', updatedAtUtc: '' };
 
 const InProgress = ({ application }: { application: any }) => {
-  const { appId, submitterId: primaryAffliation, state, expiresAtUtc, updatedAtUtc } = application;
+  const { appId, submitterId: primaryAffliation, state, expiresAtUtc, updatedAtUtc } = initState;
```
Comment (samrichca) [inline components/pages/Applications/Dashboard/Applications/InProgress/index.tsx:14]: primaryAffliation

typo!

File: components/pages/Applications/Dashboard/Applications/Start.tsx [hunk 6]
```diff
@@ -4,11 +4,27 @@ import Icon from '@icgc-argo/uikit/Icon';
...
+    fetchWithAuth({
+      url: '/applications',
```
Comment (samrichca) [inline components/pages/Applications/Dashboard/Applications/Start.tsx:19]: please use APPLICATIONS_PATH

File: components/pages/Applications/Dashboard/Applications/Start.tsx [hunk 7]
```diff
@@ -4,11 +4,27 @@ import Icon from '@icgc-argo/uikit/Icon';
...
+      method: 'POST',
+    })
+      .then(({ data }) => {
+        router.push(`/applications/${data.appId}`);
```
Comment (samrichca) [inline components/pages/Applications/Dashboard/Applications/Start.tsx:23]: APPLICATIONS_PATH here too

**Comment by ciaranschutte** [comment]
@samrichca created new const object for API endpoints constants.
internal and external should be separate constants as they're not actually the same thing despite same names.

File: global/constants/externalPaths.ts [hunk 8]
```diff
@@ -24,3 +24,8 @@ export const GLASGOW_UNI_LINK = 'https://www.gla.ac.uk/';
 export const OICR_LINK = 'https://www.oicr.on.ca/';
 export const ICGC_DCC_LINK = 'https://dcc.icgc.org/';
 export const ICGC_PCAWG_LINK = urlJoin(ICGC_DCC_LINK, 'pcawg');
+
+// API
+export const API = {
+  APPLICATIONS: '/applications',
```
Comment (samrichca) [inline global/constants/externalPaths.ts:30]: good idea

File: components/pages/Applications/Dashboard/Applications/Start.tsx [hunk 9]
```diff
@@ -4,11 +4,29 @@ import Icon from '@icgc-argo/uikit/Icon';
...
   const theme = useTheme();
+  const { fetchWithAuth } = useAuthContext();
+  const router = useRouter();
+
+  const createNewApplication = () => {
+    fetchWithAuth({
+      url: API.APPLICATIONS,
+      method: 'POST',
+    })
+      .then(({ data }: { data: any }) => {
```
Comment (samrichca) [inline components/pages/Applications/Dashboard/Applications/Start.tsx:24]: is there a type for this data? if this feature is still WIP that's fine

Comment (ciaranschutte) [reply components/pages/Applications/Dashboard/Applications/Start.tsx:24]: wip. i think we can type this at the higher fetch level. as i don't remember needing to type the `then` and `catch` vars ever before

Comment (samrichca) [reply components/pages/Applications/Dashboard/Applications/Start.tsx:24]: if i remember right, axios was giving me grief higher up, so i might be overcorrecting

fine for wip tho

**Review by samrichca — APPROVED** [review]

**Pull request merged**